datafusion/core/tests/sqllogictests/test_files/scalar.slt

```diff
@@ -646,27 +646,27 @@ SELECT CASE WHEN NULL THEN 'foo' ELSE 'bar' END
 bar
 
 # case_expr_with_null()
-query I
+query ?
```

Comment:
It's strange. I can't understand it.

I exec sql to confirm `type coercion` is right

```sql
 select arrow_typeof(case when b is null then null else b end) from (select a,b from (values (1,null),(2,3)) as t (a,b)) a;

+------------------------------------------------------------+
| arrow_typeof(CASE WHEN a.b IS NULL THEN NULL ELSE a.b END) |
+------------------------------------------------------------+
| Int64                                                      |
| Int64                                                      |
+------------------------------------------------------------+
```